### PR TITLE
[Terrain] First pass at smooth and eyedropper brush modes

### PR DESCRIPTION
**Pull request by mbalfour-amzn**
## What does this PR do?

This adds the concept of "brush modes" to the paintbrush tool, which now includes paint, eyedropper, and smooth. The brush modes are switched via different icons in the viewport during component mode. 

Eyedropper gets the current value of the data source (image gradient) underneath the cursor and sets the color/intensity/opacity of the Paint Brush Settings to match the value.

Smooth uses an average of a neighborhood of values around each current value to smooth / blur the data underneath the paintbrush. It currently uses either Gaussian, mean, or median averaging with a 3x3 kernel. In later iterations the kernel size might become configurable.

The Eyedropper and Smooth icons are placeholders until better ones can get developed, but they have their own filenames so that they can be replaced without any further code changes.

Per UX feedback, this also adds a second inner circle to the paintbrush manipulator that shows the size of the hardness setting for paint/smooth modes, and a tiny circle to show the center for paintbrush mode.

A large portion of the changes in PaintBrushManipulator.cpp is just code that's been pulled out to common functions so that it can be shared between painting and smoothing. 

## How was this PR tested?

Tried out the different modes in-Editor to verify that they performed as expected.

https://user-images.githubusercontent.com/82224783/199850802-fa4d9d87-6c4d-409e-bbd2-b0731219072f.mp4

### CODE REVIEW

**Comment by HogJonny-AMZN** [comment]
These are great additions.

**Review by HogJonny-AMZN — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]
Awesome!

File: Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.h [hunk 1]
```diff
@@ -66,28 +66,85 @@ namespace AzToolsFramework
...
-        void OnSettingsChanged(const PaintBrushSettings& newSettings) override;
+        //! Create the manipulator view(s) for the paintbrush.
+        void SetView(
+            AZStd::shared_ptr<ManipulatorViewProjectedCircle> innerCircle, AZStd::shared_ptr<ManipulatorViewProjectedCircle> outerCircle);
 
-        void MovePaintBrush(int viewportId, const AzFramework::ScreenPoint& screenCoordinates, bool isFirstPaintedPoint);
+        //! Calculate the radius for the inner and out circles for the paintbrush manipulator views based on the given brush settings.
+        //! @param settings The paint brush settings to use for calculating the two radii
+        //! @return The inner radius and the outer radius for the brush manipulator views
+        AZStd::pair<float, float> GetBrushRadii(const PaintBrushSettings& settings);
```
[the remark this answers is not in the document]
Comment (mbalfour-amzn) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.h:86]: Yeah, totally! If I don't get any other change requests for this PR I'll make the change in my next PR.

**Review by amzn-tmryan — APPROVED** [review]
This is awesome!

**Pull request merged**